### ISSUE TITLE
Spatial Anchor Expiration date not working as expected

### ISSUE THREAD

**Issue by Nuno-Monteiro** [Azure/azure-spatial-anchors-samples#273]
**Description**

When using the sample and creating a spatial anchor I can't retrieve it after a couple of hours (i'm setting the expiration to 7 days after DateTime.Now). I can't really tell where the problem is because when checking my sharing url it throws a not found error. However, checking the ASA usage I don't see any deletions so I'm sure that I'm not willingly causing this.

Is this because I have the free tier? I checked the documentation and also couldn't find any mention of this behavior.

**Steps to reproduce the issue**

1. Run the sample app and create an anchor (it will very likely by #0)
2. Go to your sharing url and confirm the anchor was created 
3. (https://your-asa-sharing-service.azurewebsites.net/api/anchors/last) or ((https://your-asa-sharing-service.azurewebsites.net/api/anchors/<anchorId>)
2. Wait a couple of hours (in my case about 7 is enough)
3. Go to your sharing url again and try to get the info of the anchor
4. (https://your-asa-sharing-service.azurewebsites.net/api/anchors/last) or ((https://your-asa-sharing-service.azurewebsites.net/api/anchors/<anchorId>)
4. See error {"type":"https://tools.ietf.org/html/rfc7231#section-6.5.4","title":"Not Found","status":404,"traceId":"|8cb384d5-4352d617a8ca7117."}

**Expected behavior**

The anchor is expected to expire after the expiry time that I set it to.

**Development information (please complete the following information)**

 - OS: macOS
 - Development Platform\Version:  Visual Studio 2019, Unity 2020.3.11f1
 - Azure Spatial Anchors SDK version : 2.9.0

**AR Device information (please complete the following information):**

 - Device: iPhone 12
 - OS: iOS 14.4

**Comment by VipulKakkad-MSFT** [comment]
Hey Nuno-Monteiro,

The deletion metrics in the Azure Portal should only show deletions, not expirations, so that is expected behavior.
The anchor expiration functionality should not be affected by pricing at all.

The ASA service uses UTC for all timestamps, so ideally, you should be using DateTime.UtcNow, instead of DateTime.Now - however, if you're trying to set an expiry 7 days out, this shouldn't be affecting your use case..

If you have an anchor ID and/or a session ID for a case where you experienced this unexpected behavior, we can investigate further what might be happening.

**Comment by Nuno-Monteiro** [comment]
Hi Vipul, thanks for the reply!

I've answered on slack but answering here as I think it's worth it. It should be specified in the documentation that the free tier service will reset itself after some inactivity (1h from my calculations) in order to keep costs down to you and as a way for free tier users to experience what it is like to use the service without any starting costs.

What I also think it's lacking is a clear tutorial on how to take the next step and how much would it cost to do it. In my specific scenario, I've put the tech to the test and experimented with it and think it's awesome. I want to take the next step, which is to have a central service I can query from any device and get any of the anchors that were previously created.
I've followed the cosmos db tutorial, but it is unclear to me how do I tie it to my sharing service and how much it would cost.
I think if this was clearer you would convert a lot more free users because the service is amazing!
Also, I'm willing to be your test case to create a more welcoming approach to devs and help out on a better tutorial that helps users transition from free tier to paid tiers.
Just let me know how can I help and what do I have to do :)
Cheers!

**Comment by Nuno-Monteiro** [comment]
TL;DR for other users that may find this issue:
The free tier sharing service that you create on Azure Portal will reset itself after inactivity (of about 1h). This means all the created anchors will be deleted and the anchor count will start over from 0 again.

**Comment by VipulKakkad-MSFT** [comment]
To clarify and expand upon what has been said above:

- The limitation of data retention due to inactivity is due to the sharing service from [this sample](https://docs.microsoft.com/en-us/azure/spatial-anchors/tutorials/tutorial-share-anchors-across-devices?tabs=azure-portal%2CVS%2CAndroid) being deployed as a free tier Azure App Service (and independent Azure resource than the Azure Spatial Anchors account).
- Azure Spatial Anchors does not have any pricing tiers at account creation at the moment, [the pricing model ](https://azure.microsoft.com/en-us/pricing/details/spatial-anchors/) only depends on usage within the month in question, and does not affect lifetime or expiry behavior of anchors.

@Nuno-Monteiro,
There is, in fact, a continuation of that tutorial [here](https://docs.microsoft.com/en-us/azure/spatial-anchors/tutorials/tutorial-use-cosmos-db-to-store-anchors), which swaps out the in-memory storage of the sample with a CosmosDb-backed persistent data store - meaning that the lifetime of the data is not tied to the free-tier Azure App Service, and persists even if the app is inactive for a while.

CosmosDb pricing (documented [here])(https://azure.microsoft.com/en-us/pricing/details/cosmos-db/) is independent of ASA pricing, since they are independent resources being used together in this scenario.

Glad to hear you're enjoying the service, and thank you for providing feedback about the samples!
Let us know if anything is still confusing, and we can clarify and/or update the samples and documentation as necessary :)
